pkg/apis/serving/v1beta1/predictor_model.go

```diff
@@ -135,14 +146,17 @@ func (m *ModelSpec) getModelTypeLabel() string {
 	return "mt:" + mt.Name
 }
 
-func getServingRuntimeSupportedModelTypeLabelSet(supportedModelTypes []v1alpha1.Framework) stringSet {
+func (m *ModelSpec) getServingRuntimeSupportedModelTypeLabelSet(supportedModelTypes []v1alpha1.ModelType) stringSet {
 	set := make(stringSet, 2*len(supportedModelTypes)+1)
 
 	// model type labels
 	for _, t := range supportedModelTypes {
-		set.add("mt:" + t.Name)
-		if t.Version != nil {
-			set.add("mt:" + t.Name + ":" + *t.Version)
+		// If runtime isn't explicitly set, only add labels for modelTypes where AutoPlace is true.
+		if m.Runtime != nil || (t.AutoPlace != nil && *t.AutoPlace) {
```

Comment:
I don't think it makes sense to have the `m.Runtime` check here since AFAICS this code path already assumes that `m.Runtime == nil`.

Reply:
I believe this check is for https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/components/predictor.go#L95 where a runtime is explicitly specified, and the given framework/version in the predictor is checked against the supportingModelTypes in the SR. Perhaps it may make sense to remove that check altogether, and just use the ServingRuntime regardless if a user explicitly specifies it? WDYT?

Reply:
@pvaneck ah yes I missed this, you're right.. sorry @Suresh-Nakkeran